### PR TITLE
Monorepo with Lerna

### PR DESCRIPTION
**Pull request by JasonRitchie**
### Reasons for making this change

Use Lerna to maintain core and themes as a monorepo. This should streamline creation of new themes and maintenance of existing theme(s).

### Status

I've published the packages to a temporary npm org using 'lerna publish'. I've also verified travis pipeline runs.

https://travis-ci.org/JasonRitchie/react-jsonschema-form/builds/607277676

https://www.npmjs.com/~jasonritchie

### Tasks before finalizing PR

- [ ] finalize package names in /packages/**/package.json files
- [ ] finalize package version #s in /packages/**/package.json files and lerna.json
- [ ] verify official travis pipeline works
- [ ] verify official publish to npm works

### Future opportunities for improvement

- [ ] Expand playground with a theme chooser
- [ ] Hoist more packages to the root of the monorepo
- [ ] Standardize build and deployment processes across packages
- [ ] Leverage travis to test themes as well as core

### CODE REVIEW

**Comment by epicfaace** [comment]
Hey, will take a look at this soon, but thanks for going ahead and doing this!

File: packages/core/.gitignore [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Created by https://www.gitignore.io/api/osx,node,linux,windows
```
[the remark this answers is not in the document]
Comment (JasonRitchie) [reply packages/core/.gitignore]: cleaned up the redundant gitignore files

**Comment by JasonRitchie** [comment]
I think there's a clear path of improvements we should pursue after we get this merged. There's good reason to merge this before it's "perfect" because integrating upstream changes when the paths of most files has changed has some friction and can leave confusing commit history.

I'll add "themes in playground" to the "future improvements" list in the initial request.

What do you think about the "Tasks before finalizing PR" list as it stands?

**Comment by epicfaace** [comment]
Thanks! I ended up moving the playground folder back to `rjsf/core` because it seems that moving it to a separate folder will require a new webpack configuration, etc. which can probably be done later. Waiting for the CI to pass and then will merge.

**Pull request merged**